tools/generate_json_rpc_docs.py

```diff
@@ -36,13 +36,14 @@ class DocumentationItem:
     def __init__(self, name, type_):
         """
         @param name: str
-        @param type: str
+        @param type_: str
         """
         self.name = name
         self.type = type_
         self.brief = DocumentationText()
         self.params = []
         self.results = []
+        self.current_tag = None
```

Comment:
This is just a "nice to have", I think. Does current_tag need to be global?

Reply:
Defining fields outside of the constructor is possible in Python but nearly all linters warn you about it, because you can end up with `AttributeError`s being thrown when things aren't always as the dev had originally thought.

Reply:
Isn't _init_ basically like a main() function? So it's still in a big scope. You didn't change anything semantically, but I believe that the style could still be improved. But not in this PR.

Reply:
No - `__init__` is the constructor(ish) for objects in Python, i.e. the class-level method responsible for _initialising_ new objects. 

Accessing fields on an object that haven't been guaranteed by the init function is generally not advised.

Reply:
Ok.